fastly/elasticsearch_test.go

```diff
@@ -0,0 +1,342 @@
+	var ess []*Elasticsearch
+	record(t, "elasticsearch/list", func(c *Client) {
+		ess, err = c.ListElasticsearch(&ListElasticsearchInput{
+			Service: testServiceID,
+			Version: tv.Number,
+		})
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(ess) < 1 {
+		t.Errorf("bad elasticsearch: %v", ess)
+	}
+
+	// Get
+	var nes *Elasticsearch
+	record(t, "elasticsearch/get", func(c *Client) {
+		nes, err = c.GetElasticsearch(&GetElasticsearchInput{
+			Service: testServiceID,
+			Version: tv.Number,
+			Name:    "test-elasticsearch",
+		})
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if es.Name != nes.Name {
+		t.Errorf("bad name: %q", es.Name)
+	}
+	if es.Format != nes.Format {
+		t.Errorf("bad format: %q", es.Format)
+	}
+	if es.Index != nes.Index {
+		t.Errorf("bad index: %q", es.Index)
+	}
+	if es.URL != nes.URL {
+		t.Errorf("bad url: %q", es.URL)
+	}
+	if es.Pipeline != nes.Pipeline {
+		t.Errorf("bad pipeline: %q", es.Pipeline)
+	}
+	if es.User != nes.User {
+		t.Errorf("bad user: %q", es.User)
+	}
+	if es.Password != nes.Password {
+		t.Errorf("bad password: %q", es.Password)
+	}
+	if es.RequestMaxEntries != nes.RequestMaxEntries {
+		t.Errorf("bad request_max_entries: %q", es.RequestMaxEntries)
+	}
+	if es.RequestMaxBytes != nes.RequestMaxBytes {
+		t.Errorf("bad request_max_bytes: %q", es.RequestMaxBytes)
+	}
+	if es.Placement != nes.Placement {
+		t.Errorf("bad placement: %q", es.Placement)
+	}
+	if es.TLSCACert != nes.TLSCACert {
+		t.Errorf("bad tls_ca_cert: %q", es.TLSCACert)
+	}
+	if es.TLSHostname != nes.TLSHostname {
+		t.Errorf("bad tls_hostname: %q", es.TLSHostname)
+	}
+	if es.TLSClientCert != nes.TLSClientCert {
+		t.Errorf("bad tls_client_cert: %q", es.TLSClientCert)
+	}
+	if es.TLSClientKey != nes.TLSClientKey {
+		t.Errorf("bad tls_client_key: %q", es.TLSClientKey)
+	}
+	if es.FormatVersion != nes.FormatVersion {
+		t.Errorf("bad format_version: %d", es.FormatVersion)
+	}
+
+	// Update
+	var ues *Elasticsearch
+	record(t, "elasticsearch/update", func(c *Client) {
+		ues, err = c.UpdateElasticsearch(&UpdateElasticsearchInput{
+			Service:  testServiceID,
+			Version:  tv.Number,
+			Name:     "test-elasticsearch",
+			NewName:  String("new-test-elasticsearch"),
+			Pipeline: String("my_new_pipeline_id"),
+		})
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if ues.Name != "new-test-elasticsearch" {
+		t.Errorf("bad name: %q", ues.Name)
+	}
+	if ues.Pipeline != "my_new_pipeline_id" {
+		t.Errorf("bad pipeline: %q", ues.Pipeline)
+	}
+
+	// Delete
+	record(t, "elasticsearch/delete", func(c *Client) {
+		err = c.DeleteElasticsearch(&DeleteElasticsearchInput{
+			Service: testServiceID,
+			Version: tv.Number,
+			Name:    "new-test-elasticsearch",
+		})
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestClient_ListElasticsearch_validation(t *testing.T) {
+	var err error
+	_, err = testClient.ListElasticsearch(&ListElasticsearchInput{
+		Service: "",
+	})
+	if err != ErrMissingService {
+		t.Errorf("bad error: %s", err)
+	}
```

Comment:
I think this should also check that the Version is not zero.

https://github.com/fastly/go-fastly/pull/164/files#diff-d244653eb384ba4e0eb5a2753701f7a3R61